src/action.c

```diff
@@ -26,85 +54,119 @@ activator_or_focused_view(struct view *activator, struct server *server)
 }
 
 void
-action(struct view *activator, struct server *server, const char *action, const char *command, uint32_t resize_edges)
+action(struct view *activator, struct server *server, struct wl_list *actions, uint32_t resize_edges)
 {
-	if (!action)
+	wlr_log(WLR_INFO, "action() start");
+
+	if (!actions) {
```

Comment:
Should this be `wl_list_empty(...)`?

Reply:
No. That one just guards against invalid internal calls which provide NULL.
If the list is empty it will just not enter the loop body.